server/src/main/java/org/candlepin/dto/api/v1/EventDTO.java

```diff
@@ -0,0 +1,525 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.candlepin.audit.Event;
+import org.candlepin.dto.CandlepinDTO;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.util.Date;
+
+/**
+ * A DTO representation of the Event entity
+ */
+@ApiModel(parent = CandlepinDTO.class, description = "DTO representing an event")
+@XmlRootElement(namespace = "http://fedorahosted.org/candlepin/Event")
+@XmlAccessorType(XmlAccessType.PROPERTY)
+public class EventDTO extends CandlepinDTO<EventDTO> {
+    public static final long serialVersionUID = 1L;
+
+
+    /**
+     * Internal DTO object for PrincipalData.
+     */
+    public static class PrincipalDataDTO {
+        private final String type;
+        private final String name;
+
+        @JsonCreator
+        public PrincipalDataDTO(
+            @JsonProperty("type") String type,
+            @JsonProperty("name") String name) {
+            if (type == null || type.isEmpty()) {
+                throw new IllegalArgumentException("type is null or empty");
+            }
+
+            if (name == null || name.isEmpty()) {
+                throw new IllegalArgumentException("name is null or empty");
+            }
+
+            this.type = type;
+            this.name = name;
+        }
+
+        public String getType() {
+            return type;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == this) {
+                return true;
+            }
+
+            if (obj instanceof PrincipalDataDTO) {
+                PrincipalDataDTO that = (PrincipalDataDTO) obj;
+
+                EqualsBuilder builder = new EqualsBuilder()
+                    .append(this.getType(), that.getType())
+                    .append(this.getName(), that.getName());
+
+                return builder.isEquals();
+            }
+
+            return false;
+        }
+
+        @Override
+        public int hashCode() {
+            HashCodeBuilder builder = new HashCodeBuilder(37, 7)
+                .append(this.getType())
+                .append(this.getName());
+
+            return builder.toHashCode();
+        }
+    }
+
+    private String id;
+    private String targetName;
+    private String principalStore;
+    private PrincipalDataDTO principal;
+    private Date timestamp;
+    private String entityId;
+    private String ownerId;
+    private String consumerId;
+    private String referenceId;
+    private String eventData;
+    private String messageText;
+    private Event.Type type;
+    private Event.Target target;
+    private Event.ReferenceType referenceType;
```

Comment:
For these, we should probably have the DTO store strings rather than passing through the event enums.